app/presenters/facet_presenter.rb

```diff
@@ -0,0 +1,181 @@
+##
+# Display logic for facets
+class FacetPresenter
+  include FacetsHelper
+  include UrlHelper
+  include ActionView::Helpers::NumberHelper
+  delegate :t, to: I18n
+
+  def initialize(facet, controller, configuration = controller.blacklight_config)
+    @facet = facet
+    @configuration = configuration
+    @controller = controller
+    @response = controller.instance_variable_get(:@response)
+  end
+
+  def display(show_children = false)
```

Comment:
Cyclomatic complexity for display is too high. [8/6]<br>Assignment Branch Condition size for display is too high. [20.25/15]<br>Method has too many lines. [13/10]<br>Perceived complexity for display is too high. [10/7]